### ISSUE TITLE
import type { PyProxy} from "pyodide/ffi" doesn't work with NodeNext

### ISSUE THREAD

**Issue by quassnoi** [pyodide/pyodide#4255]
## 🐛 Bug

```typescript
import type { PyProxy} from "pyodide/ffi"
```

doesn't work in TypeScript with `NodeNext` or `bundler` code generation (the default in Vite)

### To Reproduce

https://www.typescriptlang.org/play?target=99&moduleResolution=99&module=199#code/JYWwDg9gTgLgBDAnmApnA3nACorUIAeicAvnAGb4hwBEYiEAJsIygPTnnA0DcAUHwDGEAHYBnCABsUAOkkQA5gAp0ZAIZjsufEQCU-IA

Make sure to add pyodide to Plugins / Custom npm modules on the right hand side

### Expected behavior

No errors

### Environment

- Pyodide Version: 0.24.1
- Browser version:
- Any other relevant information:

### Additional context

Latest versions of TypeScript do honor the exports section of package.json and ffi is not listed there.

Add this to the module's `package.json/exports`:

```
"./ffi": {
      "types": "./ffi.d.ts"
    }
```

**Comment by hoodmane** [comment]
Thanks for the report @quassnoi would you be willing to make a PR?

**Comment by quassnoi** [comment]
I haven't tried to build it all the way, but when I copy my changes to the local node_modules folder, it works

**Comment by hoodmane** [comment]
I am having trouble reproducing this locally, can you give me a local reproduction? If I do:
```bash
echo "import type { PyProxy } from "pyodide/ffi";" > a.ts
npm init -y
npm i pyodide
npx tsc a.ts
```
I don't get any errors.

**Comment by quassnoi** [comment]
> I am having trouble reproducing this locally, can you give me a local reproduction? If I do:
> 
> ```shell
> echo "import type { PyProxy } from "pyodide/ffi";" > a.ts
> npm init -y
> npm i pyodide
> npx tsc a.ts
> ```
> 
> I don't get any errors.

Create a tsconfig.json with this content:

```
{
    "compilerOptions": {
        "module": "NodeNext"
    }
}
```

**Comment by quassnoi** [comment]
Just for the reference, [this](https://www.typescriptlang.org/docs/handbook/modules/reference.html#packagejson-exports) is what it is all about:

> Note that the presence of "exports" prevents any subpaths not explicitly listed or matched by a pattern in "exports" from being resolved.

**Comment by hoodmane** [comment]
Great that causes our test to fail.

**Comment by hoodmane** [comment]
I'll push to your branch if you don't mind.

**Comment by quassnoi** [comment]
> I'll push to your branch if you don't mind.

Sure please do

**Comment by hoodmane** [comment]
Closed by #4256

**Comment by hoodmane** [comment]
Thanks @quassnoi!

**Comment by Tanimodori** [comment]
It would be better if the fix is shipped in a new release. Apparently it is not included in current NPM version 0.24.1 of pyodide.
